### DISCUSSION TITLE
Requirements for a status page application

### DISCUSSION THREAD

**Discussion by fkr** [SovereignCloudStack/issues#89] [Ideas]
In Team OPS the topic of a status page application came up a while ago. Back then we concluded it would be nice if the SCS project could recommend an available status page application that can be used to assist a SCS operator with being transparent about their SCS environment towards their customers.
The various solutions that were brought up and some sort of deficiency so that none was really a _recommendation_. 
[CachetHQ](https://github.com/CachetHQ/Cachet) is the most promising one. However the project does not do regular releases, has open CVEs. The forks that exists shortly showed activity but then the activity decreased there as well.
Maybe it is worthwhile to do a joint effort to actively maintain a CachetHQ fork, maybe it is worthwhile to come up with something else (or find something else) - however, in order to judge that, it makes sense to compile the requirements for such an application.
Please use this discussion and add items / requirements ;)

**Comment by fkr** [comment]
To get this started, a few thoughts on this:

* The status page application should be simplistic in software design to not depend on a large variety of services 
* The status page applicaton should allow for simple and easy theming 

* As a CSP, I want to have a status page that allows to
  * define locations and similar grouping (AZs, ...)
  * define components globally or per location
    * to ease maintainence I want to define per component where it belongs so that I only have to define a component once, but have it visible in several locations
    * status per component should be allowed to be toggeable per location or overall
    * a component should allow for several status, that are define by me
  
* Status, Status Items should be easy to extract
  * API to interact with
    * this allows for embedding status information in other applications, such as cloud dashboards
    * this also allows for submitting items from other tooling
* As a consumer of the status page, I'd like to subscribe to events on the status page via e-mail
  * for everything
  * for specific components
* TBD: is there a need for alternate ways of pushing infos from the status page application? Example: RSS feed

* Allow for the ability to trigger webhooks upon certain events (to submit info to other systems via webhooks)
  
* As a CSP Operator, I want to be able to flag a component with a new status quick and easy
  * updating the status of a component should not be hard brainwork and minimize the possibilty 
  * updating a status should allow the CSP Operator to do that in a fashion that either pushes infos to the subscribers **or** just updates the status on the status page
  * updating the status can either be toggling the status of the component **or** can be accompanied by additional textual information.
  * When updating a status with textual information the status page application should make it easy for me as the CSP Operator to do in a way that if different people submit infos over time they are presented in a similar way (eg. the status page application should guide so that the resulting infos are presented in a identical way. Example: when updating infos of an incident over time the timeline should automatically be sorted by the status page application so that it does not depend on the Operator whether the newest info is on top or at the bottom. This is typical thing that varies if several people update items ;)

**Comment by fkr** [comment]
We had a fantastic session today on this in the Team OPS (Team 3) call. We filled an etherpad:

https://etherpad.opendev.org/p/20220602-status-page-scs

@itrich suggested to break this up into several aspects and add them as separate discussion items, I'll do that by the end of the week.
Next step is to define a MVP for this. This will happen in the next deep dive session of team OPS. Thanks everyone for the lively discussion!

Attached is a pdf export of the etherpad.

[20220602-status-page-scs-session.pdf](https://github.com/SovereignCloudStack/issues/files/8822531/20220602-status-page-scs-session.pdf)

**Comment by master-caster** [comment]
It looks like we have our requirements defined.
But there is no candidate that will fulfill all requirements by 100%.

Do we have enough information to create a user story for this?
